neuralpp/symbolic/z3_expression.py

```diff
@@ -188,6 +196,10 @@ def apply_python_callable_on_z3_arguments(python_callable: Callable,
         # arithmetic
         case operator.add:
             return arguments[0] + arguments[1]
+        case operator.sub:
+            return arguments[0] - arguments[1]
+        case operator.neg:
+            return - arguments[1]
```

Comment:
Why is this `return - arguments[1]` and not `return - arguments[0]` or `return - arguments`? What is the first argument (`arguments[0]`)?

Reply:
Nice catch! It's a bug. Fixed and added a test case which would have failed.